### PR TITLE
Feature/storing and displaying images in production

### PR DESCRIPTION
**Pull request by Ahontic**
Step 7
Images uploading/displaying for all imagable models and pages

### CODE REVIEW

File: app/admin/books.rb [hunk 1]
```diff
@@ -2,5 +2,27 @@
...
+      end
+      f.input :images, as: :file, multiple: true
+    end
+
+    f.actions
+  end
+
+  show do
+    attributes_table do
+      %i[title description price quantity category material issue_date height width depth].each do |element|
```
Comment (6aBblKuH) [inline app/admin/books.rb]: ```.each(&method(:row))```

File: app/views/books/show.html.haml [hunk 2]
```diff
@@ -4,7 +4,8 @@
     .col-sm-6
       .product-gallery
         .mb-20
-          %img.img-responsive{src: "https://www.bramstein.com/static/smashing-book-5-photo-stack.9d5d114e3e.jpg"}
+          %img.img-responsive
+            = image_tag url_for(book.images.first)
         .images.clearfix
           %a.img-link{href: "#"}
             %img{src: "https://www.bramstein.com/static/smashing-book-5-photo-stack.9d5d114e3e.jpg"}
```
Comment (6aBblKuH) [inline app/views/books/show.html.haml]: ?

File: db/schema.rb [hunk 3]
```diff
@@ -110,13 +110,26 @@
     t.index ['material_id'], name: 'index_books_on_material_id'
   end
 
+  create_table 'carts', force: :cascade do |t|
```
Comment (6aBblKuH) [inline db/schema.rb]: remove these Vietnamese 🇻🇳  flashbacks away please

**Comment by 6aBblKuH** [comment]
and check multiple images uploading, please

File: app/controllers/customers/omniauth_callbacks_controller.rb [hunk 4]
```diff
@@ -2,12 +2,19 @@
 
 class Customers::OmniauthCallbacksController < Devise::OmniauthCallbacksController
   def facebook
-    if current_customer.present?
-      current_customer.apply_omniauth(request.env['omniauth.auth'])
-      redirect_to edit_customer_registration_path, notice: 'Facebook Account Linked!'
+    # You need to implement the method below in your model (e.g. app/models/user.rb)
+    @customer = Customer.from_omniauth(request.env['omniauth.auth'])
```
Comment (6aBblKuH) [inline app/controllers/customers/omniauth_callbacks_controller.rb]: is these changes necessary ?

File: app/views/books/show.html.haml (hunk 2 above)
Comment (Ahontic) [reply app/views/books/show.html.haml]: fixed

File: app/controllers/customers/omniauth_callbacks_controller.rb (hunk 4 above)
Comment (Ahontic) [reply app/controllers/customers/omniauth_callbacks_controller.rb]: Not at all. Moved back.

File: app/admin/books.rb (hunk 1 above)
Comment (Ahontic) [reply app/admin/books.rb]: Not applicable.

File: db/schema.rb (hunk 3 above)
Comment (Ahontic) [reply db/schema.rb]: Done

File: app/admin/books.rb [hunk 5]
```diff
@@ -1,6 +1,37 @@
 # frozen_string_literal: true
 
+# rubocop:disable Metrics/BlockLength
```
Comment (6aBblKuH) [inline app/admin/books.rb]: exclude this rule for `app/admin` directory
